Changed markup for headers

rnowm opened the pull request:
New markup according to discussions during Barcelona workweek

etiennesegonzac commented:
Please don't commit `.DS_Store` files.

etiennesegonzac commented:
Please don't commit `.DS_Store` files.

fabi1cazenave commented on headers/index.html:
```diff
@@ -2,72 +2,118 @@
-      <button class="i-arrow-left"><span></span></button>
-      <h1>Mark Alfenito</h1>
-    </header>
+  <header class="app-header">
```
Aria roles should **NOT** be replaced by arbitrary class names.

As discussed in the meeting we had with the UX and FE teams and Andreas, that’s precisely what should be avoided if we want these components to be reusable.

caseyyee replied:
@fabi1cazenave as we've discussed, its not practical to do styling entirely with role attributes alone.

etiennesegonzac replied:
@caseyyee yes but there is strictly **no gain** is adding css class to already expressive markup.

fabi1cazenave replied:
@caseyyee and as we’ve discussed, classes should be used as a _complement_ of aria roles (e.g. for presentation purposes), and certainly not as a _replacement_.

It’s not a matter of taste: this condition is **required** to have generic building blocks that can be reused without getting all the drawbacks from a framework. If you don’t trust the developers on that, please remember that it’s been already discussed and decided in public.

fabi1cazenave commented on headers/index.html:
```diff
@@ -2,72 +2,118 @@
+    <menu>
+       <button><span data-icon="edit">edit</span></button>
+       <button><span data-icon="add">add</span></button>
+    </menu>
```
I don’t think `data-icon` attributes make any sense for presentation purposes.

rnowm committed:
New structure for header's markup

rnowm committed:
removed unneeded images

rnowm committed:
small markup changes

feedback from pull request:
 <header class="app-header"> changed to: <header role="toolbar">
and:
data-icon="something" to class="icon icon-something"

jmcanterafonseca commented:
role='toolbar' in a header element is totally incorrect. Please check the HTML5 spec on which it is said that if a header element has a role, it should be banner. Then we need to change this. 

We have two options

A/ 

``` html
<body role="application">
  <header>
    <menu type="toolbar">
    </menu>
  </header>
</body>
```

And put a CSS rule over the first child header element of a <body role="application"> element 

or B/ setting a class over the header. 

The toolbar role is for the menu INSIDE the header. We will make a new pull  request following this approach.

fabi1cazenave commented:
@jmcanterafonseca agreed, the markup you propose in **A/** makes much more sense for a toolbar.

Concerning the `<header>` node itself, we still want to have a CSS selector to target it. I’d be fine with `<header role="banner">` if it makes sense, but I guess a selector like `body > header` could also be used in most cases. What’s your opinion on this?

About the `<body role="application">` you’re perfectly right. I’m not pushing to get this in each and every app because it doesn’t help styling, but a `role="application"` could be used pretty much everywhere in Gaia.

jmcanterafonseca commented:
Please see  pull request #16
